pr_agent/tools/pr_code_suggestions.py

```diff
@@ -0,0 +1,124 @@
+import copy
+import json
+import logging
+import textwrap
+
+from jinja2 import Environment, StrictUndefined
+
+from pr_agent.algo.ai_handler import AiHandler
+from pr_agent.algo.pr_processing import get_pr_diff
+from pr_agent.algo.token_handler import TokenHandler
+from pr_agent.algo.utils import convert_to_markdown, try_fix_json
+from pr_agent.config_loader import settings
+from pr_agent.git_providers import get_git_provider, GithubProvider
+from pr_agent.git_providers.git_provider import get_main_pr_language
+
+
+class PRCodeSuggestions:
+    def __init__(self, pr_url: str, cli_mode=False):
+
+        self.git_provider = get_git_provider()(pr_url)
+        self.main_language = get_main_pr_language(
+            self.git_provider.get_languages(), self.git_provider.get_files()
+        )
+        self.ai_handler = AiHandler()
+        self.patches_diff = None
+        self.prediction = None
+        self.cli_mode = cli_mode
+        self.vars = {
+            "title": self.git_provider.pr.title,
+            "branch": self.git_provider.get_pr_branch(),
+            "description": self.git_provider.get_pr_description(),
+            "language": self.main_language,
+            "diff": "",  # empty diff for initial calculation
+            'num_code_suggestions': settings.pr_code_suggestions.num_code_suggestions,
+        }
```

Comment:
**Suggestion:** Consider using a data class or a named tuple for the `vars` attribute in the `PRCodeSuggestions` class. This will make the code more readable and easier to maintain.
```suggestion
        from dataclasses import dataclass

        @dataclass
        class PrVars:
            title: str
            branch: str
            description: str
            language: str
            diff: str
            num_code_suggestions: int

        self.vars = PrVars(
            title=self.git_provider.pr.title,
            branch=self.git_provider.get_pr_branch(),
            description=self.git_provider.get_pr_description(),
            language=self.main_language,
            diff="",  # empty diff for initial calculation
            num_code_suggestions=settings.pr_code_suggestions.num_code_suggestions
        )
```